src/Sparse/Row.jl

```diff
@@ -116,6 +116,22 @@ function sparse_row(R::NCRing, A::Vector{Tuple{Int, Int}}; sort::Bool = true)
   return SRow(R, A)
 end
 
+@doc raw"""
+    sparse_row(idx::Int, coeff::T; check::Bool=true) where {T}
+
+Constructs a sparse row with at most one non-zero entry `coeff` in 
+position `idx`.
+"""
+function sparse_row(idx::Int, coeff::T; check::Bool=true) where {T}
+  check && is_zero(coeff) && return SRow(parent(coeff))
+  return SRow(parent(coeff), Int64[idx], T[coeff]; check=false)
```

Comment:
Note that `check=false` is necessary here in order to not throw away the already allocated vectors. 

I do not know what to do in the case of integers. The constructor does not seem to accept a `Vector` of `ZZRingElem`s in that case, so I just defaulted this to a dummy constructor for the moment. Maybe someone with more insight can adjust the constructor below to also be performant.

Reply:
Well, see `src/Sparse/ZZRow.jl`, line 318 and following:
```julia
sparse_inner_type(::Type{ZZRingElem}) = ZZRingElem_Array

function sparse_row(R::ZZRing)
  return SRow(R, Int[], ZZRingElem_Array())
end
```
and then again line 489 and following:
``julia
for T in (Int, ZZRingElem)
  @eval begin
    function sparse_row(R::ZZRing, A::Vector{Tuple{Int, $T}}; sort::Bool = true)
      if sort && length(A) > 1
        A = Base.sort(A, lt=(a,b)->isless(a[1], b[1]))
      end
      a = ZZRingElem_Array()
...
```

Based on this I think something like this might work, in a separate `sparse_row` method for `ZZRingElem`:
```julia
  arr = ZZRingElem_Array(1)
  arr[1] = coeff
  return SRow(ZZRing, Int[idx], arr)
```